Realm/RLMRealmConfiguration.mm

```diff
@@ -373,7 +368,7 @@ - (RLMSyncConfiguration *)syncConfiguration {
     if (!_config.sync_config) {
         return nil;
     }
-    return [[RLMSyncConfiguration alloc] initWithRawConfig:*_config.sync_config];
```

Comment:
Do we still need to initialise the RLMSyncConfiguration every time we are reading the syncConfiguration, doesn't RLMSyncConfiguration holds a reference to the SyncConfig from core

Reply:
RLMRealmConfiguration does not hold onto a RLMSyncConfiguration instance, so we have to create the obj-c wrapper on each access. IIRC retaining it has awkward reference cycle issues.